### ISSUE TITLE
FormField render render a border for children input inside custom component

### ISSUE THREAD

**Issue by c3s4** [grommet/grommet#3908]
When I use a custom component, with an input field, as child of a `FormField` I have the bottom border from `FormField` and the full border from the input field.

Here an example using a composed field for password input and a plain `TextInput`

https://codesandbox.io/s/wrong-border-for-form-field-bbyuf

**Comment by myktra** [comment]
Yes! This happens when for example you want to use React Hook Form to manage your form context (for better performance on large forms, avoiding re-renders) and you wind up using their [Controller component](https://react-hook-form.com/get-started#ControlledInput) inside a FormField.

**Comment by ShimiSun** [comment]
Thanks for reporting guys, just to make sure you know you can remove the border from the theme https://v2.grommet.io/formfield#formField.border.side, but I'm aware it will affect all of your FormFields and not just the custom component one.

This issue makes me think we need to consider a `border` prop on the FormField component and not just on the theme.

**Comment by c3s4** [comment]
Yes, but is not what I need. I want to have to bottom border, but I don't want the input border.

As workaround I themed the controller border color as `transparent`. This only works because in my app I don't use any input field outside the FormField because, otherwise, all the inputs without FormField will be rendered without any border at all.

**Comment by myktra** [comment]
@ShimiSun I'd agree with your point in general, once the extra border can be removed (i.e. addressing the root cause).

FWIW, we're looking for a more traditional all-sided border around the field, so we customize it a bit via the theme.

But I don't want borders at all on my FormFields that wrap CheckBox and RadioButtonGroup (borders seem odd to include for those types of UI widgets) and want to remove extra padding as well so they are spaced uniformly from the accompanying label. The only solution that I could find (but I'm quite thankful is possible) is to extend my theme and allow for a custom `noBorder` FormField prop:

```
formField: {
  label: { ... },
  border: {
    side: 'all',
    color: 'light-4',
    size: '1px'
  },
  extend: ({ noBorder }) =>
     noBorder &&
      css`
        > label + div {
          border: none;
          > div {
            padding: 0;
          }
        }
      `
}
```

**Comment by ShimiSun** [comment]
@myktra , very helpful, thanks for sharing!

**Comment by c3s4** [comment]
@myktra this is a good workaround, better than mine, but it can't work if you are using typescript, because you can't use a not defined prop. Maybe you can "extend" the FormField component to add this prop, but I think is not very practical.

I think this issue is very near to a bug because the "style behavior" is very different just trying to create a common component as the password I wrote in the codesandbox (btw: I copied it from Grommet storybook).

**Comment by myktra** [comment]
@c3s4 good point, and it so happens I'm not (yet) using Typescript on my example project.

**Comment by forresthopkinsa** [comment]
I ended up doing this in my custom component:

```jsx
return (
	<ThemeContext.Extend
		value={{
			global: {
				control: {
					border: {
						width: '0',
					},
				},
			},
		}}
	>
		<TextInput
			type="text"
			name={name}
			value={displayValue}
			disabled={disabled}
			onChange={handleInputChange}
		/>
	</ThemeContext.Extend>
);
```

**Comment by dxmann73** [comment]
We are working around this by adding an outline to overwrite the border (which sits on the container box) using something like this with Typescript:

```
const noBorderStyle: CSSProperties = {
  outline: '1px solid white',
};
```

and 

```
    <TextInput
       ...
      plain={plain}
      style={plain ? props.style : { ...props.style, ...noBorderStyle }}
      ...
    />

```

**Comment by dxmann73** [comment]
After messing around with rendering `outline: white solid 1px` to hide the border, which doesn't really work when resizing the form fields and/or with rounded borders, the solution we came up with is using FormField.contentProps with a wrapped component like this:

`
<FormField contentProps={ border: false }>
      {props.children}
</FormField>
`

That way, we don't have to change the overall theme. Hope that helps someone.
